spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Well, here is a story about fundamental purpose of Spring Boot's auto-configuration and dependency injection. That `@ServiceActivator` bean definition is a good sample of dependency injection. And here it doesn't care from where those dependency beans come.

The auto-config is *auto* only if there is no user-defined beans. So, if you decide to declare, e.g. `MockGoogleCredentials`, the auto-config won't look into that property. But this `@ServiceActivator` will be the same and free of the inversion control machinery. 

Does it make sense to you now?